azure/services/routetables/routetables.go

```diff
@@ -72,7 +72,7 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			// currently don't support specifying your own routes via spec
 			routeTableSpec.Subnet.RouteTable.Name = to.String(existingRouteTable.Name)
 			routeTableSpec.Subnet.RouteTable.ID = to.String(existingRouteTable.ID)
-
+			s.Scope.SetSubnet(routeTableSpec.SubnetName, routeTableSpec.Subnet)
```

Comment:
sneaked in a bug fix: subnet wasn't getting updated since SubnetSpec was changed to not be a pointer in #1144